### ISSUE TITLE
Regularity for modules over quotient rings

### ISSUE THREAD

**Issue by Devlin-Mallory** [Macaulay2/M2#3321]
If S is a polynomial ring, R = S/I, and M an R-module, "regularity M" is essentially defined as "regularity betti res M". Since res M may be infinite, the resulting number is essentially only a reflection of the default LengthLimit used in "res M" (which the user also can't give as an option for regularity M). 

Here's an example illustrating this dependence; note that the homological properties of M don't depend on n at all, so whatever regularity of a module over a quotient ring *should* do, it seems that this is not what we'd want:
```
n = 6 -- make this whatever you want
S = QQ[x,y_2..y_n]
R = S/x^3
M = R^1/x^2
regularity M -- returns ceiling(n/2), because the default LengthLimit used in the resolution is n+1
```

The question then is what "should" regularity M do? I would propose that if someone asks for the regularity of an S/I-module M, what they really want is the regularity of M, viewed as an S-module via restriction of scalars. (This matches up with what people mean when they talk about the regularity of a sheaf on a variety X = Proj(S/I) embedded in projective space Proj(S).) So, regularity M could compute a resolution of M over S and read off the Betti table. 

Alternatively, regularity M could just refuse to return anything when M isn't a module over a polynomial ring. What do people think?

**Comment by mahrud** [comment]
I agree with your first suggestion, but curious to know if @mikestillman had something else in mind.

**Comment by mahrud** [comment]
@eisenbud you mentioned regularity of modules over koszul algebras is finite, but this may only be bounded by regularity of the pushforward of the module to the ambient ring. Which way does the bound go? And are the definitions of regularity based on betti numbers and local cohomology still the same on a quotient ring?
